cmake/bob_robotics.cmake

```diff
@@ -486,11 +486,6 @@ macro(BoB_build)
     # Link threading lib
     BoB_add_link_libraries(${CMAKE_THREAD_LIBS_INIT})
 
-    # Clang needs to be linked against libm and libstdc++ explicitly
-    if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
-        BoB_add_link_libraries(m stdc++)
-    endif()
-
```

Comment:
I feel we do this with GeNN too - is it not required any more?

Reply:
It doesn't seem to be. Perhaps it's been fixed in newer versions of clang? I've tested with clang on macOS and Linux and it seems fine without this.

This also forces clang to use GNU's libstdc++ rather than LLVM's libc++, which could cause problems if you don't have libstdc++. macOS was telling me that libstdc++ is deprecated so it might not work in the future.